src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -126,19 +135,29 @@ trait RegularSync {
         //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
 
-          val currentBranchTotalDifficulty = headersQueue.map(_.number)
+          val oldBranch = headersQueue.map(_.number)
             .map(blockNumber => blockchain.getBlockByNumber(blockNumber))
-            .collect{
-              case Some(b) => b.header.difficulty
-              case None => BigInt(0)
-            }.sum
+
+          val oldBlocks: Seq[(BlockBody, BigInt)] = oldBranch
+            .collect { case Some(b) => b }
+            .map{b => (b.body, b.header.difficulty)}
```

Comment:
BTW, if we wanted to retain this functionality then we could simply do `val oldBranch = headersQueue.map(_.number).flatMap(blockchain.getBlockByNumbe)`. But that's not the point.